Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1274,8 +1275,27 @@ namespace OpenInfraPlatform {
 								[&pnIndices](auto& el) -> size_t { return pnIndices ? pnIndices.get()[el - 1] - 1 : el - 1; });
 							polygon->addFace(std::begin(indices), std::end(indices));
 
-							if (face.template isOfType<typename IfcEntityTypesT::IfcIndexedPolygonalFaceWithVoids>())
-								BLUE_LOG(warning) << "Ignoring voids at " << face->getErrorLog();
+							if (face.template isOfType<typename IfcEntityTypesT::IfcIndexedPolygonalFaceWithVoids>()) {
+								auto faceWithVoids = face.template as<typename IfcEntityTypesT::IfcIndexedPolygonalFaceWithVoids>();
+
+								std::vector<std::pair<size_t, size_t>> indicesWithVoids;
+								std::vector<size_t> secondIndices(faceWithVoids->InnerCoordIndices.size());
+
+								std::transform(std::begin(faceWithVoids->InnerCoordIndices[0]), std::end(faceWithVoids->InnerCoordIndices[0]), std::begin(secondIndices),
+									[&pnIndices](auto& el) -> size_t { return pnIndices ? pnIndices.get()[el - 1] - 1 : el - 1; });
+								for (size_t i = 0; i < faceWithVoids->CoordIndex.size(); ++i)
+								{
+									indicesWithVoids[i].first = indices[i];
+								}
+
+								for (size_t i = 0; i < faceWithVoids->InnerCoordIndices.size(); ++i)
+								{
+									indicesWithVoids[i].second = secondIndices[i];
+								}

```

Comment:
This doesn't seem right. What exactly are you trying to achieve here?

Reply:
The type of the indicesWithVoids  should be `std::vector<std::pair<size_t, size_t>>`. (Same as path_all_loops in GeomUtils.cpp (Line 218)). I could not transform CoordIndices and InnerCoordIndices to the vector of the type `std::vector<std::pair<size_t, size_t>>`. So I decided to create two separate vectors and put them together. These for loops are doing this.

Reply:
Now it even more doesn't seem right. We need to discuss this.

Reply:
This code is redundant. I've forgot do delete it.

Reply:
Resolved:
e5e0196